lib/improver/spotdata_new/apply_lapse_rate.py

```diff
@@ -0,0 +1,206 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Apply temperature lapse rate adjustments to a spot data cube."""
+
+import numpy as np
+
+import iris
+from improver.spotdata_new.spot_extraction import SpotExtraction
+
+
+class SpotLapseRateAdjust:
+    """
+    Adjusts spot data temperatures by a lapse rate to better represent the
+    conditions at their altitude that may not be captured by the model
+    orography.
+    """
+
+    def __init__(self, neighbour_selection_method='nearest',
+                 grid_metadata_identifier='mosg'):
+        """
+        Keyword Args:
+            neighbour_selection_method (str):
+                The neighbour cube may contain one or several sets of grid
+                coordinates that match a spot site. These are determined by
+                the neighbour finding method employed. This keyword is used to
+                extract the desired set of coordinates from the neighbour cube.
+                The methods available will be all, or a subset, of the
+                following::
+
+                  - nearest
+                  - nearest_land
+                  - nearest_land_minimum_dz
+                  - nearest_minimum_dz
+
+                The method available in a neigbour cube will depend on the
+                options that were specified when it was created.
+            grid_metadata_identifier (str or None):
+                A string to search for in the input cube's attributes that
+                can be used to ensure that the cubes being used are for the
+                same model/grid. This test can be bypassed by setting this to
+                None.
+        """
+        self.neighbour_selection_method = neighbour_selection_method
+        self.grid_metadata_identifier = grid_metadata_identifier
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return ('<SpotLapseRateAdjust: neighbour_selection_method: {}, '
+                'grid_metadata_identifier: {}>'.format(
+                    self.neighbour_selection_method,
+                    self.grid_metadata_identifier))
+
+    def _get_attributes(self, cube):
```

Comment:
Could we call this something more general please - `_get_grid_attributes`?

Reply:
The function itself will get whatever attributes are requested, not just grid attributes, even if that it what we are using it for. Do you feel strongly about it?